app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.java

```diff
@@ -16,10 +17,14 @@
 import org.mifos.mobile.utils.Toaster;
 
 public class PassCodeActivity extends MifosPassCodeActivity {
-
+    private String actions;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        if(getIntent().getAction()!=null && !getIntent().getAction().trim().equals("android.intent.action.MAIN"))
```

Comment:
You can create a constant string inside  `util/Constants.java` for "android.intent.action.MAIN"